### PR TITLE
v0.2

### LINKED ISSUE #37: Use `%q` instead of `%s` to print paths and remove quotes from `String` methods

### PR DESCRIPTION
**Pull request by jfrohnhofen**
Multiple smaller fixes and improvements
Fixes #37 #15 #17 #28 #36 #35

### CODE REVIEW

File: RULES/cc/toolchain.go [hunk 1]
```diff
@@ -37,14 +37,14 @@ type GccToolchain struct {
...
+		includesStr.WriteString(fmt.Sprintf("-I%q ", include))
 	}
 	for _, include := range gcc.Includes {
-		includesStr.WriteString(fmt.Sprintf("-isystem %s ", include))
+		includesStr.WriteString(fmt.Sprintf("-isystem %q ", include))
 	}
 
 	return fmt.Sprintf(
-		"%s -c -o %s -MD -MF %s %s %s %s",
+		"%q -pipe -c -o %q -MD -MF %q %s %s %q",
```
Comment (ljanyst) [inline RULES/cc/toolchain.go:47]: Hah! Thanks! Does it make things faster? :)

File: cmd/build.go [hunk 2]
```diff
@@ -322,7 +344,8 @@ func copyBuildAndRuleFiles(moduleName, modulePath, buildFilesDir string, modules
 		packageName, targets := parseBuildFile(buildFile)
 		targetLines := []string{}
 		for _, targetName := range targets {
-			targetLines = append(targetLines, fmt.Sprintf("    ctx.AddTarget(reflect.TypeOf(__internal_pkg{}).PkgPath()+\"/%s\", %s)", targetName, targetName))
+			targetLines = append(targetLines,
+				fmt.Sprintf("    ctx.AddTarget(reflect.TypeOf(__internal_pkg{}).PkgPath()+\"/%s\", %s, out(\"\"))", targetName, targetName))
```
Comment (ljanyst) [inline cmd/build.go:348]: `%q`?

**Review by ljanyst — APPROVED** [review]

File: RULES/cc/toolchain.go (hunk 1 above)
Comment (jfrohnhofen) [reply RULES/cc/toolchain.go:47]: No idea, it's already so fast it's hard to notice any difference

File: cmd/build.go (hunk 2 above)
Comment (jfrohnhofen) [reply cmd/build.go:348]: Does not work, unfortunately, since we need `"/xyz"`, not `"xyz"`

**Pull request merged**